src/mongo/delegates/Views.class.php

```diff
@@ -599,8 +600,37 @@ protected function doJoins($source, $joins, &$dest, $from, $contextAlias, $build
                     {
                         // make sure any sequences are expanded before extracting properties
                         if (isset($ruleset['joins'])) $this->expandSequence($ruleset['joins'],$linkMatch);
-
-                        $dest[_GRAPHS][] = $this->extractProperties($linkMatch,$ruleset,$from);
+                        if (isset($ruleset['filter']))
+                        {
+                            foreach($ruleset['filter'] as $filterPredicate => $filter)
+                            {
+                                foreach($filter as $filterType => $filterMatch)
+                                {
+                                    if (isset($linkMatch[$filterPredicate]))
```

Comment:
The Views class has a labeller - adding this check isn't an issue - but...

How can the data be inconsistent? The $linkMatch will always contain the alias won't it? Would things work if it was the fquri? It seems to me it would have to be the alias.

And the $filterPredicate is something we're in control of when we specify the view - again which is going to be the alias.

Adding the check in is not a big thing, but how can the data be inconsistent like that?